app/lib/classification-queue.js

```diff
@@ -0,0 +1,101 @@
+import apiClient from 'panoptes-client/lib/api-client';
+import { Split } from 'seven-ten';
+
+const FAILED_CLASSIFICATION_QUEUE_NAME = 'failed-classifications';
+const MAX_RECENTS = 10;
+
+class ClassificationQueue {
+    constructor(storage, api) {
+        this.storage = storage || window.localStorage;
+        this.apiClient = api || apiClient;
+        this.recents = [];
+    }
+
+    add(classification) {
+        var queue = this._loadQueue();
+        queue.push(classification);
+
+        try {
+            this._saveQueue(queue);
+            console.info('Queued classifications:', queue.length);
+        }
+        catch(error) {
+            console.error('Failed to queue classification:', error);
+        }
+
+        this.flushToBackend();
+    }
+
+    length() {
+        return this._loadQueue().length
+    }
+
+    addRecent(classification) {
+        if (this.recents.length > MAX_RECENTS) {
+            let droppedClassification = this.recents.pop();
+            droppedClassification.destroy();
+        }
+
+        this.recents.unshift(classification);
+    }
+
+    flushToBackend() {
+        var queue = this._loadQueue();
```

Comment:
`const queue`

Reply:
But as it's sending stuff out, it's removing elements from the array. I know that we're not changing the "pointer", but marking it const seems like a lie to readers of the code?

Reply:
It's never reassigned so it's `const` in JS.